### DISCUSSION TITLE
seqr docker container not connecting to postgres container

### DISCUSSION THREAD

**Discussion by atb1r21** [broadinstitute/seqr#3488] [Q&A]
Hi,

I'm starting a new Seqr instance on a machine that I have inherited from others. I have done my best to clear out the old config and start from fresh, however I am facing an issue that the seqr container is not passing its healthcheck.

I've run docker-compose logs -f seqr:

seqr_1             | + pg_retries=10
seqr_1             | + '[' 10 -eq 10 ']'
seqr_1             | + echo 'Postgres database wasn'\''t available after 10 connection attempts'
seqr_1             | + exit 1
seqr_1             | Postgres database wasn't available after 10 connection attempts

And see the continuous attempts to connect to the postgres database container. I've checked the postgres container and it looks okay.

From the host of both containers I am able to telnet into port 5433 on the postgres container, so I was wondering if there is something wrong with the instance of seqr I have downloaded?

It is most likely not the issue, but rather an issue on the VM I am using, is there a way to troubleshoot the connectivity between containers? I have checked and they are on the same network, so am confused about why it is falling over.

Any help would be great appreciated.

Thanks,
Alister

**Comment by hanars** [comment]
Hi Alister, it is difficult to diagnose connectivity issues on your VM without having access to it and its configurations, but there are a couple things we can try. I would recommend you take everything down, and then instead of bringing up the seqr container bring up just the postgres container (`docker-compose up postgres`) and wait until thats healthy, including making sure you can connect to the database itself and not just the container. After postgres is healthy, then bring up the seqr container. If that does not work, can you please provide the log files for both the postgres and seqr containers?

**Reply by atb1r21** [reply]
Hi again,

I've tried bringing both of them up and down have checked out the various containers including the health of the database on the postgres container:

[root@hostname seqr]# docker exec -it 0c60d69ca7c7 /bin/bash

root@0c60d69ca7c7:/# pg_isready -h postgres -U postgres
postgres:5433 - accepting connections
root@0c60d69ca7c7:/# 

Sadly the seqr container is still not finding the postgres container. I have attached the logs for both containers here. Additionally I have inspected the network and the containers are being added to the same virtual bridge network, so communication should be possible.

[postgreslogs.txt](https://github.com/broadinstitute/seqr/files/12079150/postgreslogs.txt)
[seqrlog.txt](https://github.com/broadinstitute/seqr/files/12079152/seqrlog.txt)

I've checked out the logs and I can see that the postgres container is ready to accept connections but the seqr container never contacts it.

Is there a way to get a container for seqr that doesn't run the /entrypoint.sh script? The container dies after two minutes automatically and so interactive debug isn't possible.

If you have any other thoughts I'd really appreciate them, as I can't see the issue so far. I did inherit this machine from others and so am concerned that there could be issues with it regarding the previous users configuring docker.

Thanks,
Alister

**Reply by atb1r21** [reply]
Hi,

Actually I've noticed some conflicts with the RHEL firewall, is there some firewall specific setup that seqr requires?

Thanks,
Alister

**Reply by hanars** [reply]
There is no specific firewall setup that seqr requires, it requires there to be no firewall at all. I would recommend you work directly with whoever has set up your environment to make sure that the seqr and postrges containers do not have any sort of firewall blocking traffic between them. There are no seqr-specific settings you can use to configure this, so our team will unfortunately be unable to help you with this, you will need to work with the folks on your end who set up the firewall
